.github/workflows/run_tests.yaml

```diff
@@ -11,24 +11,30 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - name: checkout git repo
-      uses: actions/checkout@v2
+      uses: actions/checkout@v4
 
     - name: run tests
       shell: bash
       env:
         GITHUB_ACTIONS_CI: 1
       run: |
-        docker-compose build
-        docker-compose run re_api sh scripts/run_tests.sh
-        docker-compose down --remove-orphans
+        docker compose build
+        docker compose run re_api sh scripts/run_tests.sh
+        docker compose down --remove-orphans
+
+    - name: Upload coverage to Codecov
+      uses: codecov/codecov-action@v5
+      with:
+        token: ${{ secrets.CODECOV_TOKEN }}
+        fail_ci_if_error: true
 
   docker_build_and_push:
```

Comment:
Is this still necessary given the other workflows?

Reply:
I am not following this. Are you asking if the code coverage upload step is necessary?

Reply:
No, I'm asking if docker_build_and_push is necessary

Reply:
No services should be pulling images at all. Rancher may be pulling the image, but if you delete that chunk of the yaml the image will still be there, so next time we want to update we just switch docker to GHA

Reply:
* other than the catalog, and that's not for core services

Reply:
👍